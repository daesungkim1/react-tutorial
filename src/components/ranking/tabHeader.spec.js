import React from 'react'
import { shallow } from 'enzyme'

import renderer from 'react-test-renderer';
import RankingTabHeader from './tabHeader'

describe('<RankingTabHeader />', () => {

  it('renders correctly', () => {

    const wrapper = shallow(
      <RankingTabHeader
        nth={3}
        name={'tab name here'}
        onTabChange={()=>true} />
    )

    // should have correct class value with text
    expect(wrapper.find('a.link_tab3').text())
      .toBe('tab name here')
  })

  it('render', () => {
    const tree = renderer.create(
      <RankingTabHeader
        nth={3}
        name={'tab name here'}
        onTabChange={()=>true} />
    ).toJSON()

    // manually execute onMouseOver to get a snapshot
    tree.props.onMouseOver(tree.props.name)
    expect(tree).toMatchSnapshot()
  })

});