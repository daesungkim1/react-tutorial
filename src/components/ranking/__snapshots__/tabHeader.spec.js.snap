exports[`<RankingTabHeader /> render 1`] = `
<a
  className="link_tab link_tab3"
  href="#none"
  onMouseOver={[Function]}>
  tab name here
</a>
`;
